tests/bluetooth/audio/cap_commander/src/test_broadcast_reception.c

```diff
@@ -0,0 +1,442 @@
+{
+	struct cap_commander_test_broadcast_reception_fixture *fixture = f;
+
+	bt_cap_commander_unregister_cb(&mock_cap_commander_cb);
+
+	for (size_t i = 0; i < ARRAY_SIZE(fixture->conns); i++) {
+		mock_bt_conn_disconnected(&fixture->conns[i], BT_HCI_ERR_REMOTE_USER_TERM_CONN);
+	}
+}
+
+static void cap_commander_test_broadcast_reception_teardown(void *f)
+{
+	free(f);
+}
+
+static void init_start_param(void *f)
+{
+	struct cap_commander_test_broadcast_reception_fixture *fixture = f;
+	int err;
+
+	fixture->start_param.type = BT_CAP_SET_TYPE_AD_HOC;
+	fixture->start_param.param = fixture->start_member_params;
+
+	fixture->start_param.count = ARRAY_SIZE(fixture->start_member_params);
+
+	for (size_t i = 0; i < ARRAY_SIZE(fixture->start_subgroups); i++) {
+		fixture->start_subgroups[i].bis_sync = 1 << i;
+#if defined(CONFIG_BT_AUDIO_CODEC_CFG_MAX_METADATA_SIZE)
+		/* TODO: add tests where metadata_len > 0 */
+		fixture->start_subgroups[i].metadata_len = 0;
+#else
+		fixture->start_subgroups[i].metadata_len = 0;
+#endif /* CONFIG_BT_AUDIO_CODEC_CFG_MAX_METADATA_SIZE */
+	}
+
+	for (size_t i = 0U; i < ARRAY_SIZE(fixture->start_member_params); i++) {
+		fixture->start_member_params[i].member.member = &fixture->conns[i];
+		bt_addr_le_copy(&fixture->start_member_params[i].addr, BT_ADDR_LE_ANY);
+		fixture->start_member_params[i].adv_sid = SID;
+		fixture->start_member_params[i].pa_interval = ADV_INTERVAL;
+		fixture->start_member_params[i].broadcast_id = BROADCAST_ID;
+		memcpy(fixture->start_member_params[i].subgroups, &fixture->start_subgroups[0],
+		       sizeof(struct bt_bap_bass_subgroup) * NR_SUBGROUPS);
+		fixture->start_member_params[i].num_subgroups = NR_SUBGROUPS;
+	}
+
+	for (size_t i = 0; i < ARRAY_SIZE(fixture->conns); i++) {
+		err = bt_cap_commander_discover(&fixture->conns[i]);
+		zassert_equal(0, err, "Unexpected return value %d", err);
+	}
+
+	for (size_t i = 0U; i < ARRAY_SIZE(fixture->start_member_params); i++) {
+		fixture->start_member_params[i].member.member = &fixture->conns[i];
+	}
+}
+
+ZTEST_SUITE(cap_commander_test_broadcast_reception, NULL,
+	    cap_commander_test_broadcast_reception_setup,
+	    cap_commander_test_broadcast_reception_before,
+	    cap_commander_test_broadcast_reception_after,
+	    cap_commander_test_broadcast_reception_teardown);
+
+ZTEST_F(cap_commander_test_broadcast_reception, test_commander_reception_start)
+{
+	int err;
+
+	init_start_param(fixture);
+
+	err = bt_cap_commander_register_cb(&mock_cap_commander_cb);
+	zassert_equal(0, err, "Unexpected return value %d", err);
+
+	err = bt_cap_commander_broadcast_reception_start(&fixture->start_param);
+	zassert_equal(0, err, "Unexpected return value %d", err);
+
+	zexpect_call_count("bt_cap_commander_cb.broadcast_reception_start", 1,
+			   mock_cap_commander_broadcast_reception_start_cb_fake.call_count);
+}
+
+ZTEST_F(cap_commander_test_broadcast_reception, test_commander_reception_start_one_subgroup)
```

Comment:
What does this test that `test_commander_reception_start` doesn't?

Reply:
Test with one subgroup instead of CONFIG_BT_BAP_BASS_MAX_SUBGROUPS

Reply:
Right, but what if we modify `CONFIG_BT_BAP_BASS_MAX_SUBGROUPS` to be e.g. `4`?
Would you do tests for 1, 2 and 3 as well? 

If you want to test all possible values, then it should have been a loop over the values, rather than just a single test for a specific count. 

My original point was whether this will actually hit any code that wasn't already covered by a test?

Reply:
added a comment

Reply:
> Right, but what if we modify `CONFIG_BT_BAP_BASS_MAX_SUBGROUPS` to be e.g. `4`? Would you do tests for 1, 2 and 3 as well?
> 
> If you want to test all possible values, then it should have been a loop over the values, rather than just a single test for a specific count.
> 
> My original point was whether this will actually hit any code that wasn't already covered by a test?

We could loop, but I think it is sufficient to test for illegal values (0 and CONFIG_BT_BAP_BASS_MAX_SUBGROUPS+1), maximum, and minimum.